integration-tests/utils.ts

```diff
@@ -145,16 +146,25 @@ const expectSearchResultsToFulfillExpectation = async (
     search: { url: string; params?: any },
     bundleEntryExpectation: jest.Expect,
 ) => {
-    console.log('Searching with params:', search);
-    await expect(
-        (async () => {
-            return (
-                await client.get(search.url, {
-                    params: search.params,
-                })
-            ).data;
-        })(),
-    ).resolves.toMatchObject({
+    console.log('GET Searching with params:', search);
+    const searchResult = (await client.get(search.url, { params: search.params })).data;
+    expect(searchResult).toMatchObject({
+        resourceType: 'Bundle',
+        entry: bundleEntryExpectation,
+    });
+
+    console.log('POST Searching with params as x-www-form-urlencoded in body:', search);
+    const postSearchResult = (await client.post(`${search.url}/_search`, qs.stringify(search.params))).data;
+    expect(postSearchResult).toMatchObject({
+        resourceType: 'Bundle',
+        entry: bundleEntryExpectation,
+    });
+
+    console.log('POST Searching with same params in body and in query:', search);
+    const postSearchRepeatingParamsResult = (
+        await client.post(`${search.url}/_search`, qs.stringify(search.params), { params: search.params })
+    ).data;
+    expect(postSearchRepeatingParamsResult).toMatchObject({
```

Comment:
I think that we should write this as a separate test case.

Running this for ALL test cases will be slow. Plus, this is not an equivalent search request to the ones above. Duplicating params MAY have a different meaning. Doing POST for ALL test cases is already doubling the execution time for this tests, but that's probably ok for now.

I think it's better to just add a couple test cases that send some query params + some params in the body.

Reply:
Emm, yea that makes sense. I'll update it. 

(Out of curiosity, why were you up and giving comments 4hrs ago? That's awfully early for you. )